VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java

```diff
@@ -34,7 +35,7 @@ public CoursesAsyncLoader(Context context, Bundle args){
     public AsyncTaskResult<List<EnrolledCoursesResponse>> loadInBackground() {
 
         Api api = new Api(getContext());
-        if (this.oauthToken == null){
+        if (Config.getInstance().getSocialSharingConfig().isEnabled() && this.oauthToken == null) {
```

Comment:
Ah okay. So this code isn't really where courses are loaded, it's where friends courses are loaded. In that case, we shouldn't be triggering this at all - though it still shouldn't be causing a failure if we do. It seems like the fix listed will cause a problem when the social sharing features are enabled for users that aren't logged in with Facebook. It may be that's a problem we should deal with when we enable that feature, but maybe there's a better fix for now.

I think it should also be fixed if we put an ``if(Config.getInstance().getSocialSharingConfig().isEnabled)`` around this code in ``MyCourseListTabFragment.java``?

```
        Bundle args = new Bundle();
        args.putString(CoursesAsyncLoader.TAG_COURSE_OAUTH, FacebookSessionUtil.getAccessToken());
        getLoaderManager().restartLoader(MY_COURSE_LOADER_ID, args, this);
```


If that doesn't work, then we should just use the fix from this bug.